### PR TITLE
[TASK] Revise form protection

### PR DESCRIPTION
**Pull request by brotkrueml**
- Move important note from the bottom to the top
- Add a TOC
- Prepend PHP namespaces with a backslash
- Use admonitions
- Remove superfluous version information
- Do some formatting

Releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 1]
```diff
@@ -1,150 +1,167 @@
...
+reasons.
 
 When processing the data that has been submitted by the form,
 you can check that the form token is valid like this:
 
-.. code-block:: php
+..  code-block:: php
+
+    if ($dataHasBeenSubmitted &&
+        \TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()->validateToken(
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:73]: by using use statements the text would be much easier readable

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 2]
```diff
@@ -1,150 +1,167 @@
...
-will always return :code:`FALSE` for the second call.
+..  note::
+    The :php:`validateToken()` method invalidates the token with the token ID.
+    So calling the validation with the same parameters twice in a row will
+    always return :php:`false` for the second call.
 
-It is important that the tokens get validated **before** the tokens are persisted.
-This makes sure that the tokens that get invalidated by :code:`validateToken`
-cannot be used again.
+..  important::
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:90]: do not use "important" as it is displayed green like a note. Use .. attention instead

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 3]
```diff
@@ -1,150 +1,167 @@
...
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:148]:
```suggestion
..  attention::
```

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 4]
```diff
@@ -154,20 +171,19 @@ Usage is the same as in backend context:
...
-cannot be used again.
+..  note::
+    The :php:`validateToken()` invalidates the token with the token ID. So,
+    calling the validation with the same parameters twice in a row will always
+    return :php:`false` for the second call.
 
-Note that this **requires a logged on user** whether in frontend or backend. CSRF protection
-is not supported for anonymous users. Without a logged on user the token will always be
-:code:`dummyToken`. See :forge:`77403` for details.
+..  important::
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:186]:
```suggestion
..  attention::
```

File: Documentation/ApiOverview/FormProtection/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/FormProtection/Index.rst:73]: Agreed.

**Review by linawolf — APPROVED** [review]

**Pull request merged**